hacspec/src/poly.rs

```diff
@@ -267,38 +267,118 @@ pub(crate) fn extended_euclid_invert<T: TRestrictions<T>>(x: T, n: T, signed: bo
     t
 }
 
-/// Extended euclidean algorithm to compute the inverse of x in yℤ[x]
-#[inline]
-pub fn extended_euclid<T: TRestrictions<T>>(x: &[T], y: &[T], n: T) -> Result<Vec<T>, &'static str> {
-    let (x, y) = normalize(x, y);
-    println!("euclid: {:?} / {:?}", x, y);
-
-    let mut new_t = vec![T::default(); x.len()];
-    new_t[0] = T::from_literal(1);
+// Pad to degree d and reverse coefficients
+fn reverse<T: TRestrictions<T>>(x: &[T], d: usize) -> Vec<T> {
+    let mut out = vec![T::default(); d+1];
+    for i in 0..std::cmp::min(x.len(), d+1) {
+        out[d-i] = x[i]; //XXX: this could probably be written more nicely with iterators?
```

Comment:
You can just use `reverse` on the vector (something like `x.to_vec().reverse().truncate(d)`).
https://doc.rust-lang.org/std/vec/struct.Vec.html#method.reverse